documentation/modules/ROOT/pages/02-architecture.adoc

```diff
@@ -1,104 +1,255 @@
+This effectively avoids the need to keep the applications backward compatible with all the events ever emitted.
+Introducing breaking changes in the event schema is as easy as making sure to consume all the events of given type from the _Event Mesh_.
+This also works for the systems which can't have downtime windows.
+The applications could have short-lived backward compatible layers for such situations.
+When all the events are processed, the backward compatible code may be removed simplifying the maintenance.
 
-=== Creating  tables
- 
-[cols="1a,1a,1a"]
+Because in the long-term, the regular datastores are the source of truth for the system, all traditional techniques for application maintenance apply well.
+It is also, easier to understand for developers, as it avoids sophisticated event handlers logic, and reconciling into the read database abstraction.
+
+=== Differences from the _Service Mesh_
+
+Worth pointing out are the differences from the _Service Mesh_ pattern.
+The _Service Mesh_ pattern is intended to improve the resilience of synchronous communications which return the response.
+The _Service Mesh_ effectively raises the uptime of the dependent endpoints by retrying and backoff policies.
+The uptime can't be raised to 100%, though, so _Service Mesh_ still can lose the messages.
+
+The table below captures the key differences:
+
+[cols=".^1,.^2,.^2"]
 |===
-| *Column A*  | Column *A* | _Column C_
-|
-* Lorem Ipsum
-* Lorem Ipsum
+|| Service Mesh | Event Mesh
 
 |
-* Lorem Ipsum
-* Lorem Ipsum
+*Similarities*
+
+2+a|
+
+* Flexibility
+* Robustness
+* Decoupling
 
 |
-* Lorem Ipsum
-* Lorem Ipsum
+*Differences*
+a|
+* Synchronous
+* Request and response
+* Better for queries
+
+a|
+* Asynchronous
+* Events
+* Better for commands
+
 |===
 
+=== Supporting Legacy Systems
+
+One of the strengths of an _Event Mesh_ architecture is its ability to integrate seamlessly with legacy systems, making them more resilient and adaptable.
+Legacy applications can be retrofitted to produce and consume events through lightweight adapters.
+For instance:
+
+* A monolithic legacy application can send events for specific operations, instead of handling all logic internally in transactional fashion.
 
+* Event listeners can be introduced incrementally, enabling the legacy app to subscribe to events without refactoring its core logic.
 
-=== Content that can be copied
+* This approach decouples old systems from rigid workflows, allowing for gradual modernization while ensuring operational continuity.
 
-Click below to copy the content
-[.console-input]
-[source,shell script]
-----
-oc version #openshift cli client
-oc login --token=<token> --server=<server>
-----
+=== Improving Resilience in Applications
 
+Traditional systems often rely on synchronous calls and transactions, which can cascade failures across components.
+Replacing these with asynchronous event-driven communication reduces dependencies and makes the system _Eventually Consistent_.
```

Comment:
maybe add a link to the definition of _eventual consistency_ (like a Wikipedia page)

Reply:
I didn't find any RH page for eventual consistency, and IDK if we should link to other websites. Maybe that's ok?!?

Reply:
if you don't find a good link that you can use, then maybe don't italicize but just use "eventually consistent", so that it is not really a concept that the user is not aware of.

Reply:
I'm going to link this: https://en.wikipedia.org/wiki/Eventual_consistency